services/checks/registry.go

```diff
@@ -87,28 +83,21 @@ func (r *Registry) CreateAlert(id string, labels, annotations map[string]string,
 
 	r.alerts[id] = alert
 	if r.times[id].IsZero() {
-		r.times[id] = r.nowF().Add(delayFor)
+		r.times[id] = r.nowF()
 	}
 }
 
-// RemovePrefix removes all alerts with given ID prefix except a given list of IDs.
-func (r *Registry) RemovePrefix(prefix string, keepIDs map[string]struct{}) {
+// clean clears all alerts from the registry
+func (r *registry) clean() {
 	r.rw.Lock()
 	defer r.rw.Unlock()
+	r.alerts = make(map[string]ammodels.PostableAlert)
+	r.times = make(map[string]time.Time)
```

Comment:
You don't need that `times` map without delays at all. In `collect` method you can return all alerts that you have.

Reply:
done